src/main/java/com/exasol/adapter/request/parser/PushdownSqlParser.java

```diff
@@ -114,10 +114,9 @@ public SqlNode parseExpression(final JsonObject expression) {
 
     private SqlStatementSelect parseSelect(final JsonObject select) {
         // FROM clause
-        final SqlNode table = parseExpression(select.getJsonObject("from"));
-        assert (table.getType() == SqlNodeType.TABLE) || (table.getType() == SqlNodeType.JOIN);
+        final SqlNode from = parseExpression(select.getJsonObject("from"));
         // SELECT list
-        final SqlSelectList selectList = parseSelectList(select.getJsonArray("selectList"));
+        final SqlSelectList selectList = parseSelectList(select.getJsonArray("selectList"), from);
```

Comment:
This violates the SoC. You can't have a method called "parse..." that takes an additional parameter that has nothing to do with the part being parsed.
Separate parsing and construction of the select list, please.

This would then look something like this:

```suggestion
        final SqlSelectList originalSelectList = parseOrigianlSelectList(select.getJsonArray("selectList"));
        final SqlSelectList expandedSelectList = createSelectList(originalSelectList, from);
```

Reply:
Restructured code